Geometry/RPCGeometryBuilder/src/RPCGeometryBuilderFromDDD.h

```diff
@@ -33,7 +33,7 @@ class RPCGeometryBuilderFromDDD
 
  private:
   RPCGeometry* buildGeometry(DDFilteredView& fview, const MuonDDDConstants& muonConstants);
-  std::map<RPCDetId,std::list<RPCRoll *> > chids;
+  std::map<RPCDetId,std::set<RPCRoll *> > chids;
```

Comment:
On 6/23/16 12:46 AM, Junghwan John Goh wrote:
> In Geometry/RPCGeometryBuilder/src/RPCGeometryBuilderFromDDD.h
> <https://github.com/cms-sw/cmssw/pull/14848#discussion_r68189341>:
> 
>> @@ -33,7 +33,7 @@ class RPCGeometryBuilderFromDDD
>>  
>>   private:
>>    RPCGeometry* buildGeometry(DDFilteredView& fview, const MuonDDDConstants& muonConstants);
>> -  std::map<RPCDetId,std::list<RPCRoll *> > chids;
>> +  std::map<RPCDetId,std::set<RPCRoll *> > chids;
> 
> So this will re-order the rolls inside a chamber, changing ordering
> defined in DDD/DB to the raw Id index.
> Therefore the index mapping of some rolls are swapped in the validation
> module.
> 
> There will be no cost to roll back to list. I can make a PR to fix it if
> you agree.

Since this is the most likely place for the randomness I observe,
please update.
Thank you.



> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/14848/files/ec40e5725ae281b91161ed4964375cb034430c8a#r68189341>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AEdcbkxBE9XlPmYOOn57gMBWMs4xSE4Eks5qOjnugaJpZM4IzkB0>.
>